File: src/Controls/KarveControls/DataGrid/DataGridHelpers/DataGridTables.cs
using System.Collections;
using System.Collections.Generic;
using System.Linq;

namespace KarveControls.DataGrid.DataGridHelpers
{
    public class DataGridTables : Dictionary<object, DataGridTable>
    {

        public void Add(DataGridTable TablaQuery)
        {
            TablaQuery.Item = this.Count;
            base.Add(TablaQuery.Name, TablaQuery);
        }

        public ArrayList ToArray()
        {
            dynamic HC = (from c in Values orderby c.Item select  c);
            ArrayList RS = new ArrayList();
            foreach (DataGridTable ctr in HC) {
                RS.Add(ctr);
            }
            return RS;
        }

    }
}

File: documentation/Karve/CustomControls/DataGrid/Clases/DataGridTable/DataGridTables.vb
Public Class DataGridTables
    Inherits System.Collections.Hashtable

    Public Overloads Sub Add(ByVal TablaQuery As DataGridTable)
        TablaQuery.Item = Me.Count
        MyBase.Add(TablaQuery.Name, TablaQuery)
    End Sub

    Public Function ToArray() As ArrayList
        Dim HC = (From c As DataGridTable In Me.Values Order By c.Item Select c)
        Dim RS As New ArrayList
        For Each ctr As DataGridTable In HC
            RS.Add(ctr)
        Next
        Return RS
    End Function

End Class
